docs/mocksupport.rst

```diff
@@ -0,0 +1,131 @@
+
+    classes = conn.EnumerateClasses();
+    . . .
+
+Generally this mock environment supports:
+
+1. All of the WBEMConnection request methods that communicate with the WBEM
+server.
+
+2. Multiple namespaces.
+
+3. TODO
+
+It does this by mocking WBEM Server responses at the common level below the
+WBEMConnection methods such as GetClass, etc. but before the requests are
+formatted into HTTP.  This allows a mock repository to built by the user
+into which the data is inserted for any gived test.
+
+It does NOT support:
+
+1. CIM/XML protocol security
+2. Dynamic WBEM Server providers
+3. It does not support processing of the queries defined for ExecQuery in
+languages like CQL and WQL.  The mocker just returns found instances.
+4. It does not support the filter processing of the FQL filter query language
+   used by the Open... operations.  It returns the same data as if the filter
+   did not exist.
+5. It does not support time delay for operations.  The responses are returned
+   as soon as the mocker processes the data in the repository.
+
+.. _`Adding Data to the Fake Repository`:
+
+Adding Data to the Fake Repository
+----------------------------------
+
+There are three methods in the fake WBEMConnection to add data to the
+mock repository
+
+1. Add pywbem CIM Objects directly to the repository.
+This is the method FakedWBEMConnection.add_cimobjects().
+
+2. Compile pywbem objects from strings with the pywbem MOF compiler from strings.
+This is the method FakedWBEMConnection.compile_mof_str().
+
+3. Compile pywbem objects into the fake repository from files.
+This is the method FakedWBEMConnection.compile_mof().
+
+
+.. _`FakedWBEMConnection.add_cimobjects()`:
+
+FakedWBEMConnection.add_cimobjects()
+++++++++++++++++++++++++++++++++++++
+
+This method allows you to contstruct CIM objects such as CIMClasses,
+CIMInstances, and CIMQualifierDeclarations directly and to insert them into
+the repository.
+
+.. code-block:: python
+
+    import pywbem
+    import pywbem_mock
+
+    tst_namespace = 'root/blah'
+    conn = pywbem_mock.FakedWBEMConnection()
+
+    # build the qualifier declarations
+    q1 = pywbem.CIMQualifierDeclaration('FooQualDecl1', 'uint32')
+    q2 = pywbem.CIMQualifierDeclaration('FooQualDecl2', 'string',
+                                        value='my string')
+
+    conn.add_cimobjects([q1. q2], tst_namespace)
+
+    # test the WBEMConnection GetQualifier and EnumerateQualifiers methods
+    rtn_q1 = conn.GetQualifier('FooQualDecl1', namespace=tst_namespace)
+
+    q_rtn = conn.EnumerateQualifiers(namespace=ns)
+
+
+.. _`FakeWBEMConnection.compile_mof_str()`:
+
+
+FakeWBEMConnection.compile_mof_str()
+++++++++++++++++++++++++++++++++++++
+
+.. _`FakeWBEMConnection.compile_mof()`:
+
+
+FakeWBEMConnection.compile_mof()
+++++++++++++++++++++++++++++++++
+
+
+
+.. _`FakedWBEMConnection`:
+
+
+FakedWBEMConnection
+-------------------
+
+.. automodule:: pywbem_mock._session
```

Comment:
Having the module description at this point is like another introduction of the mock support.

Reply:
TODO

Reply:
I agree but want to put off until we get some other doc cleaner.

Reply:
Now shown in list of discussion points, below.